backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -87,9 +105,38 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
                 return;
             }
 
+            if (!await LocalizationService.RobotIsLocalized(robot.Id))
```

Comment:
We have many checks to see that everything is in order, should we group these checks in a function?